docs/Deprecating-Disabling-and-Removing-Formulae.md

```diff
@@ -0,0 +1,98 @@
+# Deprecating, Disabling, and Removing Formulae
+
+There are many reasons why formulae may be deprecated, disabled, or removed. This document explains the differences between each method as well as explaining when one method should be used over another.
+
+## Overview
+
+This general rule of thumb can be followed:
+
+- `deprecate!` should be used for formulae that _should_ no longer be used.
+- `disable!` should be used for formulae that _cannot_ be used.
+- Formulae that have are not longer acceptable in homebrew/core or have been disabled for over a year should be removed.
+
+## Deprecation
+
+If a user attempts to install a deprecated formula, they will be shown a warning message but the install will succeed.
+
+A formula should be deprecated to indicate to users that the formula should not be used and may be disabled in the future. Deprecated formulae should still be able to build from source and their bottles should continue to work.
+
+The most common reasons for deprecation are when the upstream project is deprecated, unmaintained, or archived.
+
+To deprecate a formula, add a `deprecate!` call. This call should include a deprecation date (in the ISO 8601 format) and a deprecation reason:
+
+```ruby
+deprecate! date: "YYYY-MM-DD", because: :reason
+```
+
+The `date` parameter should be set to the date that the project or version became (or will become) deprecated. If there is no clear date but the formula needs to be deprecated, use today's date. If the `date` parameter is set to a date in the future, the formula will not become deprecated until that date. This can be useful if the upstream developers have indicated a date where the project or version will stop being supported.
+
+The `because` parameter can be set to a preset reason (using a symbol) or a custom reason. See the [Deprecate and Disable Reasons](#deprecate-and-disable-reasons) section below for more details about the `because` parameter.
+
+## Disabling
+
+If a user attempts to install a disabled formula, they will be shown an error message and the install will fail.
+
+A formula should be disabled to indicate to users that the formula cannot be used and will be removed in the future. Disabled formulae may no longer be able to build from source or have working bottles.
+
+The most common reasons for disabling are when the formula cannot be built from source (meaning no bottles can be built), has been deprecated for a long time, the upstream repository has been removed, or the project has no license.
```

Comment:
The parallelism in this sentence uses "when the formula":
1. cannot be built from source
2. has been deprecated for a long time
3. the upstream repository has been removed
4. the project has no license

so that doesn't quite work. You can fix it with the following, but I don't quite like it either.

```suggestion
The most common reasons for disabling are when the formula cannot be built from source (meaning no bottles can be built), the formula has been deprecated for a long time, the upstream repository has been removed, or the project has no license.
```

Reply:
Definitely better 👍

Reply:
Alternatively:

```suggestion
The most common reasons for disabling are when the formula cannot be built from source (meaning no bottles can be built), has been deprecated for a long time, had its upstream repository removed, or has no license.
```

or

```suggestion
The most common reasons for disabling are when the formula cannot be built from source (meaning no bottles can be built), has been deprecated for a long time, no longer has an upstream repository, or has no license.
```

Reply:
Is there a reason to avoid bulleted lists here?

>The most common reasons for disabling are:
>* the formula cannot be built from source (meaning no bottles can be built)
>* it has been deprecated for a long time
>* no upstream repository
>* no license

Reply:
Nope, that's just how I wrote it. Makes sense to me to have it as a list. Good suggestion.